V shazilms/updatedoc (#11011)
MicrosoftDocs/msteams-docs: msteams-platform/bots/calls-and-meetings/calls-meetings-bots-overview.md

|                                       | Choose microphone for calls                                                                                     |
|                                       |-----------------------------------------------------------------------------------------------------------------|
|                                       | Show state of a participant [Idle, Early media, Connecting, Connected, On hold, In Lobby, Disconnected]         |
|                                       |-----------------------------------------------------------------------------------------------------------------|
|                                       | Show state of a call [Early Media, Incoming, Connecting, Ringing, Connected, Hold, Disconnecting, Disconnected] |
|                                       |-----------------------------------------------------------------------------------------------------------------|
|                                       | Show if a participant is muted                                                                                  |

> |                                       |-----------------------------------------------------------------------------------------------------------------|

Reviewer: ```suggestion
```

> |                                       | Show the reason why a participant left a call                                                                   |

Reviewer: ```suggestion
```

> |---------------------------------------|-----------------------------------------------------------------------------------------------------------------|

Reviewer: ```suggestion
```

| Screen sharing | - Share the entire screen from within the app <br>- Share a specific application (from the list of running apps)<br>- Share a web browser tab from the list of open tabs<br>- Share system audio during screen sharing<br>- Participant can view remote screen share |
| Roster | - List participants <br>- Remove a participant |
| PSTN | - Place a one-to-one call with a PSTN participant <br>- Place a group call with PSTN participants<br>- Upgrade a one-to-one call with a PSTN participant into a group call<br>- Dial-out from a group call as a PSTN participant<br>- Support for early media |
| General | Test your mic, speaker, and camera with an audio testing service (available by calling 8:echo123) |
| Device management | - Ask for permission to use audio or audio and video <br>- Get camera list<br>- Set camera<br>- Get selected camera<br>- Get microphone list<br>- Set microphone<br>- Get selected microphone<br>- Get speakers list<br>- Set speaker<br>- Get selected speaker |
| Video rendering | - Render single video in many places (local camera or remote stream) <br>- Set or update scaling mode<br>- Render remote video stream |

## Set up firewalls

> Setting up firewalls is crucial to protect your network and system from unauthorized access and malicious attacks. Firewalls act as a barrier, controlling incoming and outgoing traffic based on security rules, preventing hackers from accessing your system, blocking malicious traffic, and hiding internal IP addresses. They provide an additional layer of security, preventing attacks like hacking, malware, and DoS attacks, and protect sensitive data from breaches. For more information, see https://cdn.graph.office.net/prod/pdf/teams-to-bot-communication-final.pdf.

Reviewer: ```suggestion
Setting up firewalls is crucial for protecting your network and system from unauthorized access and malicious attacks. Firewalls act as barriers, controlling incoming and outgoing traffic based on security rules. They prevent hackers from accessing your system, block malicious traffic, and hide internal IP addresses.

In the context of Microsoft Teams, setting up firewalls for calling and meeting bots is particularly important to protect sensitive communication data and prevent unauthorized access. Firewalls can help prevent toll fraud and denial-of-service (DoS) attacks on your bots, minimizing the risk of financial losses and disruption of critical communication services.
```

> In the context of Microsoft Teams, setting up firewalls for calling and meeting bots is particularly important to protect sensitive communication data and prevent unauthorized access. Firewalls can help prevent toll fraud and denial-of-service (DoS) attacks on your bots, minimizing the risk of financial losses and disruption of critical communication services.

Reviewer: ```suggestion
```

> For more information, see https://cdn.graph.office.net/prod/pdf/teams-to-bot-communication-final.pdf.

Reviewer: ```suggestion
```

Reviewer: @v-shazilms, the doc PM has confirmed that we cannot use the provided PDF. Could you please suggest an alternative link, preferably from Microsoft Learn documentation?

Now, you must understand some core concepts, terminology, and conventions.

> 

Reviewer: @v-shazilms, we need to move both **Microsoft Graph calling and meeting APIs** and **Set up firewalls** sections to before the **Code sample** section.

| General | Test your microphone, speaker, and camera with an audio testing service (available by calling 8:echo123). |
| Device management | &bull; Ask for permission to use audio or audio and video. <br> &bull; Get camera list. <br> &bull; Set camera. <br> &bull; Get selected camera. <br> &bull; Get microphone list. <br> &bull; Set microphone. <br> &bull; Get selected microphone. <br> &bull; Get speakers list. <br> &bull; Set speaker. <br> &bull; Get selected speaker. |
| Video rendering | &bull; Render single video in multiple places (local camera or remote stream). <br> &bull; Set or update scaling mode. <br> &bull; Render remote video stream. |

To get the Microsoft.Graph.Communications.Calls NuGet package, see [NuGet Gallery | Microsoft.Graph.Communications.Calls 1.2.0.10563](https://www.nuget.org/packages/Microsoft.Graph.Communications.Calls/1.2.0.10563). 

> ## Set up firewalls

Reviewer: ```suggestion
```

> Setting up firewalls is crucial for protecting your network and system from unauthorized access and malicious attacks. In the context of Teams, setting up firewalls for calling and meeting bots is important to protect sensitive communication data and prevent unauthorized access. Firewalls can help prevent toll fraud and denial-of-service (DoS) attacks on your bots, minimizing the risk of financial losses and disruption of critical communication services.

Reviewer: ```suggestion
```

| Mid-call control | &bull; Turn the video on or off. <br> &bull; Mute or unmute the microphone. <br> &bull; Switch between cameras. <br> &bull; Put the call on hold or resume locally. <br> &bull; Active speaker. <br> &bull; Choose speaker for calls. <br> &bull; Choose microphone for calls. <br> &bull; Show the state of a participant, such as idle, early media, connecting, connected, on hold, in lobby, and disconnected. <br> &bull; Show the state of a call, such as early media, incoming, connecting, ringing, connected, on hold, disconnecting, or disconnected. <br> &bull; Show if a participant is on mute. <br> &bull; Show the reason why a participant left a call. |
| Screen sharing | &bull; Share the entire screen from within the app. <br> &bull; Share a specific application (from the list of running apps). <br> &bull; Share a web browser tab from the list of open tabs. <br> &bull; Share system audio during screen sharing. <br> &bull; Participant can view remote screen share. |
| Roster | &bull; List participants. <br> &bull; Remove a participant. |

> | Public Switched Telephone Network (PSTN) | &bull; Place a one-to-one call with a PSTN participant. <br> &bull; Place a group call with PSTN participants. <br> &bull; Upgrade a one-to-one call with a PSTN participant into a group call. <br> &bull; Dial-out from a group call as a PSTN participant. <br> &bull; Support for early media. |

Reviewer 2: Most of the capabilities are covered under [call resource](https://learn.microsoft.com/en-us/graph/api/resources/call?view=graph-rest-1.0) and the associated endpoints

> | Public Switched Telephone Network (PSTN) | &bull; Place a one-to-one call with a PSTN participant. <br> &bull; Place a group call with PSTN participants. <br> &bull; Upgrade a one-to-one call with a PSTN participant into a group call. <br> &bull; Dial-out from a group call as a PSTN participant. <br> &bull; Support for early media. |

Reviewer: @samarthasthana, I couldn't find anything related to "early media" in the link mentioned above.

| General | Test your microphone, speaker, and camera with an audio testing service (available by calling 8:echo123). |

> | Device management | &bull; Ask for permission to use audio or audio and video. <br> &bull; Get camera list. <br> &bull; Set camera. <br> &bull; Get selected camera. <br> &bull; Get microphone list. <br> &bull; Set microphone. <br> &bull; Get selected microphone. <br> &bull; Get speakers list. <br> &bull; Set speaker. <br> &bull; Get selected speaker. |

Reviewer 2: @v-shazilms we can skip adding the capability for `General`, `Device management` and `Video Rendering` from this change as they are beyond Calling workload.

* To learn about Microsoft Graph SDK and its libraries, see [Microsoft Graph SDK overview](https://learn.microsoft.com/graph/sdks/sdks-overview?context=graph%2Fapi%2Fbeta&view=graph-rest-beta&preserve-view=true).
* To get the Microsoft.Graph.Communications.Calls NuGet package, see [NuGet Gallery | Microsoft.Graph.Communications.Calls 1.2.0.10563](https://www.nuget.org/packages/Microsoft.Graph.Communications.Calls/1.2.0.10563). 
* For samples that use Microsoft Graph SDK, see [Microsoft Graph Communications samples](https://github.com/microsoftgraph/microsoft-graph-comms-samples/tree/master/Samples). 

## Microsoft Graph calling and meeting APIs

> Bots can perform the following functions by calling Microsoft Graph APIs for calls and online meetings:

Reviewer 2: This is the relevant link for the graph workload for calling and online meetings.
https://learn.microsoft.com/en-us/graph/api/resources/communications-api-overview?view=graph-rest-1.0

## See also

* [Build bots for Teams](../what-are-bots.md)
* [Work with the cloud communications API in Microsoft Graph](/graph/api/resources/communications-api-overview)
* [Microsoft Graph SDK overview](/graph/sdks/sdks-overview)
* [Add Graph permissions](registering-calling-bot.md#add-graph-permissions)

> * [Develop calling and online meeting bots on your local PC](debugging-local-testing-calling-meeting-bots.md)

Reviewer 2: Question for this and the next line removed, those docs still exist, do we not need to link them anymore? Are they not relevant?

Reviewer: @samarthasthana, these links either exist in the article or appear among the list of links in the `Documentation` section (on the right-hand side of the screen). Hence, I have removed them.
